### DISCUSSION TITLE
Block or allow Internet access via Context Menu (Windows Firewall).

### DISCUSSION THREAD

**Discussion by fabcard** [farag2/Sophia-Script-for-Windows#296] [Ideas]
Hello there, @farag2 
Is it possible to add an option to block or allow programs internet access via context menu (right click)?
The idea is to easy block or allow a program to access the internet in a couple clicks.
Thanks in advance.

**Comment by farag2** [comment]
I think, this is possible but what for?..

**Comment by fabcard** [comment]
Hello friend,
Well, the idea as I said is to easily block-allow programs from access the internet.
Let's say I want to block a program xyz.exe so it doesn't access the internet, then I right-click on it and choose block. After I block, it will appear on Windows Firewall list as shown below:
![Untitled](https://user-images.githubusercontent.com/88634350/142071622-884a6026-af59-41c4-a6bf-b17f9e1d0ec0.jpg)
Same if I want to allow (unblock) it.
Here is an idea:
![Untitled2](https://user-images.githubusercontent.com/88634350/142072614-a5785c8b-0c9b-496d-a85d-37a9e3554791.jpg)
On the Sophia script could be an option to enable or disable this tweak just like everything else.
Thank you again.

**Comment by farag2** [comment]
As I can see you're already installed an app to expand the context menu (which one?). I do not want to such feature to Sophia, but this is simple to write a code for you to add to the module and preset file.

**Reply by fabcard** [reply]
Don't worry, farag2.
Feel free to delete this request.
Again, thank you.

**Reply by farag2** [reply]
Actually, I'm interested in this feature but I don't want to mix all available OS features in the ine app. I'm thinking that it may be probably a feature for a feature. 🤔
